### PR TITLE
ENG-2600 Refactoring MFE/Widgets OOTB testing

### PR DESCRIPTION
**Pull request by jeffgo10**
- decoupled widgets CRUD and OOTB widget testing
 - utilizing api controllers + amended methods
 - fixes on the drag+drop widget values

### CODE REVIEW

File: ui-tests/cypress/integration/pages/widgets-ootb.spec.js [hunk 1]
```diff
@@ -0,0 +1,1096 @@
...
+
+  const selectPageFromSidebar = (pageOpen = THE_PAGE) => {
+    const currentPageContent = currentPage.getContent();
+    currentPageContent.clickSidebarTab(1);
+    cy.wait(3000);
+    currentPageContent.selectPageFromSidebarPageTreeTable(pageOpen.code);
+    currentPageContent.clickSidebarTab(0);
+  };
+
+  describe('CMS Content Widget', () => {
```
Comment (g-di-martino) [inline ui-tests/cypress/integration/pages/widgets-ootb.spec.js:125]: The tests in this describe depend on the first test which add the widget to the page, hence they're going to fail if the first one is not execute or if it failed; you should remove the widget at the end of the first test and add it, via API, at the beginning of each other test

File: ui-tests/cypress/integration/pages/widgets-ootb.spec.js [hunk 2]
```diff
@@ -0,0 +1,1096 @@
...
+                  .should('have.class', 'PageStatusIcon--published')
+                  .and('have.attr', 'title').should('eq', 'Published');
+      cy.wrap(WIDGET_FRAME.frameNum).as('widgetToRemoveFromPage');
+      cy.wrap({ code: CMS_WIDGETS.CONTENT.code, group: 'free' }).as('widgetToRevert');
+      cy.wrap(SAMPLE_DUPE_WIDGET_CODE).as('widgetToDelete');
+    });
+
+  });
+
+  describe('CMS Content List Widget', () => {
```
Comment (g-di-martino) [inline ui-tests/cypress/integration/pages/widgets-ootb.spec.js:247]: Same as the previous describe

File: ui-tests/cypress/integration/pages/widgets-ootb.spec.js [hunk 3]
```diff
@@ -0,0 +1,1096 @@
...
+                  .should('have.class', 'PageStatusIcon--published')
+                  .and('have.attr', 'title').should('eq', 'Published');
+
+      cy.wrap([WIDGET_FRAME.frameNum, WIDGET_FRAME_2.frameNum]).as('widgetToRemoveFromPage');
+      cy.wrap(2).as('recentContentsToUnpublish');
+      cy.wrap(2).as('recentContentsToDelete');
+    });
+  });
+
+  describe('CMS Content Search Query Widget', () => {
```
Comment (g-di-martino) [inline ui-tests/cypress/integration/pages/widgets-ootb.spec.js:571]: Same as above

File: ui-tests/cypress/integration/pages/widgets-ootb.spec.js [hunk 4]
```diff
@@ -0,0 +1,1096 @@
...
+                  .should('have.class', 'PageStatusIcon--published')
+                  .and('have.attr', 'title').should('eq', 'Published');
+      cy.wrap(WIDGET_FRAME.frameNum).as('widgetToRemoveFromPage');
+      cy.wrap({ code: CMS_WIDGETS.CONTENT_QUERY.code, group: 'free' }).as('widgetToRevert');
+      cy.wrap(SAMPLE_DUPE_WIDGET_CODE).as('widgetToDelete');
+    });
+
+  });
+
+  describe('CMS Search Form and Search Results Widgets', () => {
```
Comment (g-di-martino) [inline ui-tests/cypress/integration/pages/widgets-ootb.spec.js:684]: Same

File: ui-tests/cypress/integration/pages/widgets-ootb.spec.js [hunk 5]
```diff
@@ -0,0 +1,1096 @@
...
+      cy.validateUrlPathname(`/widget/detail/${CMS_WIDGETS.SEARCH_RESULT.code}`);
+      cy.wrap([WIDGET_FRAME_1.frameNum, WIDGET_FRAME_2.frameNum]).as('widgetToRemoveFromPage');
+      cy.wrap([
+        { code: CMS_WIDGETS.SEARCH_FORM.code, group: 'free' },
+        { code: CMS_WIDGETS.SEARCH_RESULT.code, group: 'free' },
+      ]).as('widgetToRevert');
+    });
+  });
+
+  describe('CMS News Archive and News Latest Widgets', () => {
```
Comment (g-di-martino) [inline ui-tests/cypress/integration/pages/widgets-ootb.spec.js:775]: Same

**Review by g-di-martino — CHANGES_REQUESTED** [review]

**Comment by jeffgo10** [comment]
hi @g-di-martino all tests are now independent from each other. pls review

**Review by g-di-martino — APPROVED** [review]

**Pull request merged**